lib/utils/ignore.ts

```diff
@@ -294,45 +332,57 @@ export async function filterFilesWithDockerignore(
 	const dockerignoreServiceDirs: string[] = multiDockerignore
 		? Object.keys(ignoreByDir).filter((dir) => dir && dir !== root)
 		: [];
+	const files = await listFiles(projectDir);
+
 	const dockerignoreFiles: FileStats[] = [];
 	const filteredFileList: FileStats[] = [];
-	const klaw = await import('klaw');
-
-	await new Promise((resolve, reject) => {
-		// Looking at klaw's source code, `preserveSymlinks` appears to only
-		// afect the `stats` argument to the `data` event handler
-		klaw(projectDir, { preserveSymlinks: false })
-			.on('error', reject)
-			.on('end', resolve)
-			.on('data', (item: { path: string; stats: Stats }) => {
-				const { path: filePath, stats } = item;
-				// With `preserveSymlinks: false`, filePath cannot be a symlink.
-				// filePath may be a directory or a regular or special file
-				if (!stats.isFile()) {
+	await Promise.all(
+		files.map(async (filePath) => {
+			const relPath = path.relative(projectDir, filePath);
+
+			// .dockerignore files are always added to a list of known dockerignore files
+			if (path.basename(relPath) !== '.dockerignore') {
```

Comment:
```suggestion
			if (path.basename(relPath) === '.dockerignore') {
```
Some test cases are failing and I think the errors can be explained by the logic above:

```
         "[Warn] The following .dockerignore file(s) will not be used:"
         "[Warn] * /tmp/build/29c50989/versioned-source/tests/test-data/projects/no-docker-compose/basic/src/start.sh"
         "[Warn] * /tmp/build/29c50989/versioned-source/tests/test-data/projects/no-docker-compose/basic/src/windows-crlf.sh"
```

`start.sh` and `windows-crlf.sh` are not `.dockerignore` files!

Reply:
Oops, this is a typo, fixed!